features/custom_codecs/README.md

```diff
@@ -0,0 +1,379 @@
+BoundStatement bs2 = ps.bind()
+    .setInt(0, 42)
+    .set(1, myPojo, MyPojo.class); //newly-created set() method
+```
+
+And here is how to retrieve a `MyPojo` object converted from a JSON document:
+
+```java
+ResultSet rows = session.execute(...);
+Row row = one();
+// Let the driver convert the string for you...
+MyPojo myPojo = row.get(1, MyPojo.class); // newly-created get() method
+// ... or retrieve the raw string if you need it
+String json = row.get(1, String.class); // row.getString(1) would have worked too
+```
+
+Tip: `Row` and `BoundStatement` have several methods to set and retrieve values. 
+But if you plan to use custom codecs,
+make sure that you use preferably the new `get()` and `set()` methods: these new methods 
+avoid any ambiguity by requiring the user to explicitly specify the desired Java type,
+thus forcing the driver to pick the right codec for the right task. 
+
+### Support for parameterized types
+
+The new `get()` and `set()` methods can even be used to retrieve collections or other
+user-defined parameterized types; simply use the `get()` and `set()` variants that
+take a Guava's [TypeToken][TypeToken] parameter:
+
+```java
+Row row = ...
+List<MyPojo> myPojos = row.get("jsonList", new TypeToken<List<MyPojo>>(){}); 
+// or alternatively, the following also works for collections:
+List<MyPojo> myPojos = row.getList("jsonList", MyPojo.class); 
+```
+
+### Support for collections, user types, tuples and Java Enums
+
+`CodecRegistry` instances not only store default codecs and user-defined codecs;
+they can also create new codecs on the fly, based on the set of codecs they currently hold.
+
+* Codecs for [UserType][UserType] instances are created on the fly using [UDTCodec][UDTCodec];
+* Codecs for [TupleType][TupleType] instances are created on the fly using [TupleCodec][TupleCodec];
+* Codecs for Java Enums are created on the fly using [EnumStringCodec][EnumStringCodec];
+* Codecs for collections of Enums, user types and tuple types are created on the fly using
+    [ListCodec][ListCodec], [SetCodec][SetCodec] and [MapCodec][MapCodec].
+
+This way, the user does not have to manually register all derived codecs for a given "base" codec.
+However, other combinations of Java and CQL types not listed above cannot have their codecs created on the fly;
+such codecs must be manually registered.
+
+The driver also provides another codec for Enums that converts Enum instances to CQL ints 
+representing their ordinal position: [EnumIntCodec][EnumIntCodec]. If you plan to use this codec, make sure to register it
+manually, otherwise Enums would be converted using [EnumStringCodec][EnumStringCodec].
+_It is not possible to mix both codecs in the same `CodecRegistry`_.
+
+Note that the default set of codecs has no support for
+[Cassandra custom types][AbstractType];
+to be able to deserialize values of such types, you need to manually register an appropriate codec.
+
+### Using custom codecs with user types and tuples
```

Comment:
The mapper generates more efficient UDT codecs (that go directly from byte stream to target object, without the intermediary `UDTValue`). We should mention it here.